src/ctap/mod.rs

```diff
@@ -342,6 +342,7 @@ where
         if let Some(auth_param) = &pin_uv_auth_param {
             // This case was added in FIDO 2.1.
             if auth_param.is_empty() {
+                let _ = (self.check_user_presence)(cid);
```

Comment:
We do a UP check but we don't care about the result. Seems odd.

Reply:
The reason is that we return a specific error code right after.

> If authenticator supports either pinUvAuthToken or clientPin features and the platform sends a zero length pinUvAuthParam, wait for user touch and then return either CTAP2_ERR_PIN_NOT_SET if PIN is not set or CTAP2_ERR_PIN_INVALID if PIN has been set.

The reason for this behaviour:

> This is done for the case where multiple authenticators are attached to the platform and the platform wants to enforce pinUvAuthToken semantics, but the user has to select which authenticator to get the pinUvAuthToken from.

Reply:
But if the user doesn't validate UP (i.e. timeout) the authenticator hasn't really been selected but yet will return one of the 2 error codes.
I would expect that the authenticator should return that UP timed out in such case. Maybe something to clarify on the spec.

Reply:
Discussion opened for clarification on the [spec](https://github.com/fido-alliance/fido-2-specs/issues/1092) (link for those with permissions). I'll keep this PR open until resolved.

Reply:
https://github.com/fido-alliance/fido-2-specs/pull/1096
Actually, the user touch shouldn't be ignored. Spec PR on the way. Changed in the next commit.